### PR TITLE
[HP-81] Completed feature 'Basic Enemy Ship and Spawning' with tests

### PR DESCRIPTION
**Pull request by MichaelRRyan**
Additionally moved stray script files to the scripts folder.

### CODE REVIEW

File: Assets/Scripts/DoorDetector.cs [hunk 1]
```diff
@@ -1,109 +1,109 @@
...
-            level.SetActive(false);
-
-        // Enables the levels to.
-        foreach (GameObject level in levelsToEnable)
-            level.SetActive(true);
-
-        if (player) player.transform.position = movePlayerTo;
-    }
-}
+﻿//using System.Collections;
```
Comment (JackMalone1) [inline Assets/Scripts/DoorDetector.cs:1]: Can this be completely removed instead of just commented out?

File: Assets/Scripts/EnemyBehaviour.cs [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
 {
-    public float maxFireInterval = 10.0f;
 
     public const float TIME_BETWEEN_ACTIONS = 1.0f;
 
-    // The time at which the last action occurred.
-    private float _lastMovementChange;
+    public float fireInterval = 2.0f;
+    public float waveAmount = 5.0f;
+    public string enemyType;
```
Comment (JackMalone1) [inline Assets/Scripts/EnemyBehaviour.cs:12]: Should this have a default or is it not necessary?

File: Assets/Scripts/EntityManager.cs [hunk 3]
```diff
@@ -5,28 +5,31 @@
...
+	public float enemySpawnIncrease = 0.01f; // The decrease in seconds of the spawn delay every second.
+	public float spawnMargin = 1f;
+	public float spawnSpacing = 1.0f;
+	public float approxEnemyRadius = 1.5f;
+	public float spawnDistance = 5.0f; // The distance over which enemies spawn, none spawn after this distance.
+
 	public GameObject[] enemyPrefabs;
 
 	[Header("References")]
+	public Camera _camera;
```
Comment (JackMalone1) [inline Assets/Scripts/EntityManager.cs:20]: Should this start with _ if it's public?

**Review by JackMalone1 — COMMENTED** [review]
Looks pretty good, just have a look at the comments to see if you need to do anything for any of them, if not, you're good to merge

File: Assets/Scripts/DoorDetector.cs (hunk 1 above)
Comment (MichaelRRyan) [reply Assets/Scripts/DoorDetector.cs:1]: Hoping to use it later, but it doesn't need fixing just yet

File: Assets/Scripts/EntityManager.cs (hunk 3 above)
Comment (MichaelRRyan) [reply Assets/Scripts/EntityManager.cs:20]: Yes, but then it shadows the built in "camera" variable

**Pull request merged**